src/alert/lsst.rs

```diff
@@ -695,15 +774,36 @@ impl AlertWorker for LsstAlertWorker {
         let mut prv_candidates_doc = prv_candidates
             .unwrap_or(vec![])
             .into_iter()
-            .map(|x| mongify(&x))
-            .collect::<Vec<_>>();
+            .map(|mut x| {
+                if let Err(e) = x.add_mag_data() {
+                    return Err(e);
+                }
+                Ok(mongify(&x))
+            })
+            .collect::<Result<Vec<_>, _>>()?;
```

Comment:
Nice use of `collect` to convert a vec of results into a result of a vec. It's a handy trick! We can also write this as,

```rust
            .map(|mut x| {
                x.add_mag_data()?;
                Ok(mongify(&x))
            })
            .collect::<Result<Vec<_>, AlertError>>()?;
```

And similarly for `fp_hist_doc` and `non_detections_doc`.

Reply:
I like your suggestion better! Let me change that